### PR TITLE
tvs_002 : Created endpoints for Sign up and sign in with jwt authentication and cors

### PR DESCRIPTION
**Pull request by KadirSheikh**

### CODE REVIEW

File: tvs_microservices/userservice/config.toml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# filename: config.toml
+[database]
```
Comment (Shilpa2022Git) [inline tvs_microservices/userservice/config.toml:2]: don't use service port, auth information in database section. Prepare different sections as per skeleton

Comment (Shilpa2022Git) [inline tvs_microservices/userservice/config.toml:2]: don't use service port, auth information in database section. Prepare different sections as per skeleton

**Review by nkulkarni10 — APPROVED** [review]

File: tvs_microservices/userservice/model/user.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	Fullname  string `gorm:"type:varchar(50)" json:"fullname"`
+	Createdby uint64 `gorm:"type:bigint(11)" json:"createdby"`
+	DOB       string `gorm:"type:date" json:"DOB"`
+	Emailid   string `gorm:"uniqueIndex;type:varchar(30)" json:"emailid"`
+	Mobileno  string `gorm:"type:varchar(30)" json:"mobileno"`
+	Password  string `gorm:"->;<-;not null" json:"-"`
+	Token     string `gorm:"-" json:"token,omitempty"`
+}
+
+type Address struct {
```
[the remark this answers is not in the document]
Comment (KadirSheikh) [reply tvs_microservices/userservice/model/user.go:21]: Changed Address to Useraddress

File: tvs_microservices/userservice/model/user.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+	AddVehicle(vehicle UservehiclesDTO) Uservehicles
+}
+
+type UserRepository interface {
+	RegisterUser(user User) User
+	VerifyCredential(emailid string, password string) interface{}
+	IsDuplicateEmail(emailid string) (tx *gorm.DB)
+	IsDuplicateUsername(loginid string) (tx *gorm.DB)
+	IsDuplicateVehicleRegNo(vehregno string) (tx *gorm.DB)
+	AddVehicle(vehicle Uservehicles) Uservehicles
```
[the remark this answers is not in the document]
Comment (KadirSheikh) [reply tvs_microservices/userservice/model/user.go:79]: Changed RegisterUser to AddUser and IsDuplicate methods to Check.

File: tvs_microservices/userservice/service/jwt-service.go [hunk 4]
```diff
@@ -0,0 +1,69 @@
+package service
```
[the remark this answers is not in the document]
Comment (KadirSheikh) [reply tvs_microservices/userservice/service/jwt-service.go:1]: Added config, jwt in common module name utils

**Comment by KadirSheikh** [comment]
Merging PR#1 , PR#4 , PR#6 to main.

**Pull request merged**